python/src/opendp/extras/sklearn/decomposition/__init__.py

```diff
@@ -180,139 +180,191 @@ def __init__(
-        from sklearn.utils.extmath import stable_cumsum, svd_flip # type: ignore[import]
-        from sklearn.decomposition._pca import _infer_dimension # type: ignore[import]
-
-        self.mean_, S, Vt = values
-        U = Vt.T
-        n_samples, n_features = self.n_samples, self.n_features_in_
-        n_components = self.n_components  # type: ignore[attr-defined]
-
-        # CODE BELOW THIS POINT IS FROM SKLEARN
-        # flip eigenvectors' sign to enforce deterministic output
-        U, Vt = svd_flip(U, Vt)
-
-        components_ = Vt
-
-        # Get variance explained by singular values
-        explained_variance_ = (S**2) / (n_samples - 1)
-        total_var = np.sum(explained_variance_)
-        explained_variance_ratio_ = explained_variance_ / total_var
-        singular_values_ = S # Store the singular values. 
-
-        # Postprocess the number of components required
-        if n_components == "mle":
-            n_components = _infer_dimension(explained_variance_, n_samples)
-        elif 0 < n_components < 1.0:
-            # number of components for which the cumulated explained
-            # variance percentage is superior to the desired threshold
-            # side='right' ensures that number of features selected
-            # their variance is always greater than n_components float
-            # passed. More discussion in issue: https://github.com/scikit-learn/scikit-learn/pull/15669
-            explained_variance_ratio_np = explained_variance_ratio_
-            ratio_cumsum = stable_cumsum(explained_variance_ratio_np)
-            n_components = np.searchsorted(ratio_cumsum, n_components, side="right") + 1
-
-        # Compute noise covariance using Probabilistic PCA model
-        # The sigma2 maximum likelihood (cf. eq. 12.46)
-        if n_components < min(n_features, n_samples):
-            self.noise_variance_ = np.mean(explained_variance_[n_components:])
-        else:
-            self.noise_variance_ = 0.0
-
-        self.n_samples_ = n_samples
-        self.components_ = components_[:n_components, :]
-        self.n_components_ = n_components
-        self.explained_variance_ = explained_variance_[:n_components]
-        self.explained_variance_ratio_ = explained_variance_ratio_[:n_components]
-        self.singular_values_ = singular_values_[:n_components]
-
-        return U, S, Vt
-
-    def measurement(self) -> Measurement:
+        ...
+
+    def measurement(self) -> Measurement:  # type: ignore[empty-body]
         """Return a measurement that releases a fitted model."""
-        return self._prepare_fitter() >> (lambda _: self)
+        ...
 
-    # overrides an sklearn method
     def _validate_params(*args, **kwargs):
-        pass
+        ...
 
 
 _decomposition = import_optional_dependency('sklearn.decomposition', False)
 if _decomposition is not None:
-    PCA = type('PCA', (_decomposition.PCA,), dict(PCA.__dict__))  # type: ignore[assignment,misc]
+    PCAStub = PCA
```

Comment:
`PCAStub` doesn't seem to be referenced? Is this left over from an earlier attempt?